firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/impl/TestUtils.java

```diff
@@ -95,6 +101,26 @@ static void awaitAsyncOperations(ExecutorService executorService) throws Interru
     shadowOf(getMainLooper()).idle();
   }
 
+  static List<UpdateProgress> awaitProgressEvents(UpdateTask updateTask, int count)
+      throws InterruptedException {
+    List<UpdateProgress> progressEvents = new ArrayList<>();
+    updateTask.addOnProgressListener(FirebaseExecutors.directExecutor(), progressEvents::add);
+    ExecutorService executor = TestOnlyExecutors.blocking();
+    executor.execute(
+        () -> {
+          while (progressEvents.size() < count) {
+            try {
+              Thread.sleep(100);
+            } catch (InterruptedException e) {
+              throw new RuntimeException("Interrupted while waiting for progress events", e);
+            }
+          }
+        });
+    executor.awaitTermination(100, TimeUnit.MILLISECONDS);
```

Comment:
Isn't 100 ms too short? The thread sleeps 100 ms for every iteration.

Reply:
Ah, true. Adjusted to 50ms sleeps and a 500ms timeout.